### PR TITLE
[Merged by Bors] - feat: `Matrix.fromRows` and `Matrix.fromColumns` multiplied by vectors

### PR DESCRIPTION
**Pull request by madvorak**
Zulip discussion:
https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/.E2.9C.94.20sumType_zeroFun_dotProduct

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Matrix/ColumnRowPartitioned.lean [hunk 1]
```diff
@@ -144,6 +144,16 @@ section Semiring
...
 
+@[simp]
+lemma fromRows_mulVec (A₁ : Matrix m₁ n R) (A₂ : Matrix m₂ n R) (v : n → R) :
+    fromRows A₁ A₂ *ᵥ v = Sum.elim (A₁ *ᵥ v) (A₂ *ᵥ v) := by
+  ext (_ | _) <;> rfl
+
+@[simp]
+lemma vecMul_fromColumns (B₁ : Matrix m n₁ R) (B₂ : Matrix m n₂ R) (v : m → R) :
+    v ᵥ* Matrix.fromColumns B₁ B₂ = Sum.elim (v ᵥ* B₁) (v ᵥ* B₂) := by
+  ext (_ | _) <;> rfl
```
Comment (eric-wieser) [inline Mathlib/Data/Matrix/ColumnRowPartitioned.lean:155]: Can you also add the ones where the `Sum.elim` is on the LHS (and the rows/columns are swapped)?

Comment (madvorak) [reply Mathlib/Data/Matrix/ColumnRowPartitioned.lean:155]: I didn't know how to prove the other two.

Comment (eric-wieser) [reply Mathlib/Data/Matrix/ColumnRowPartitioned.lean:155]: Where do you get stuck? The proof should be trivial still.

Comment (madvorak) [reply Mathlib/Data/Matrix/ColumnRowPartitioned.lean:155]: I committed a sorried lemma. Either provide a proof, or delete it from this PR.

Comment (eric-wieser) [reply Mathlib/Data/Matrix/ColumnRowPartitioned.lean:155]: I pushed a proof; can you add the other lemma?

File: Mathlib/Data/Matrix/ColumnRowPartitioned.lean [hunk 2]
```diff
@@ -144,14 +144,31 @@ section Semiring
...
 variable [Semiring R]
 
+@[simp]
+lemma fromRows_mulVec (A₁ : Matrix m₁ n R) (A₂ : Matrix m₂ n R) (v : n → R) :
+    fromRows A₁ A₂ *ᵥ v = Sum.elim (A₁ *ᵥ v) (A₂ *ᵥ v) := by
+  ext (_ | _) <;> rfl
+
+@[simp]
+lemma vecMul_fromColumns (B₁ : Matrix m n₁ R) (B₂ : Matrix m n₂ R) (v : m → R) :
+    v ᵥ* Matrix.fromColumns B₁ B₂ = Sum.elim (v ᵥ* B₁) (v ᵥ* B₂) := by
```
Comment (eric-wieser) [inline Mathlib/Data/Matrix/ColumnRowPartitioned.lean:154]:
```suggestion
    v ᵥ* fromColumns B₁ B₂ = Sum.elim (v ᵥ* B₁) (v ᵥ* B₂) := by
```

File: Mathlib/Data/Matrix/ColumnRowPartitioned.lean (hunk 1 above)
Comment (madvorak) [reply Mathlib/Data/Matrix/ColumnRowPartitioned.lean]: Thanks!

**Comment by eric-wieser** [comment]
bors merge

Thanks for this!

**Pull request closed without merging**